webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/precomputed/PrecomputedCompressorFactory.scala

```diff
@@ -1,16 +1,28 @@
 package com.scalableminds.webknossos.datastore.datareaders.precomputed
 
-import com.scalableminds.webknossos.datastore.datareaders.{Compressor, JpegCompressor, NullCompressor}
+import com.scalableminds.webknossos.datastore.datareaders.precomputed.PrecomputedDataType.PrecomputedDataType
+import com.scalableminds.webknossos.datastore.datareaders.{
+  CompressedSegmentationCompressor,
+  Compressor,
+  JpegCompressor,
+  NullCompressor
+}
 
 object PrecomputedCompressorFactory {
   private val nullCompressor = new NullCompressor
 
-  def create(encoding: String): Compressor =
+  def create(encoding: String,
+             dataType: PrecomputedDataType,
+             chunkSize: Array[Int],
+             compressedSegmentationBlockSize: Option[Array[Int]]): Compressor =
     encoding.toLowerCase match {
-      case "raw"                     => nullCompressor
-      case "jpeg"                    => new JpegCompressor
-      case "compressed_segmentation" => throw new NotImplementedError
-      case _                         => throw new IllegalArgumentException(s"Chunk encoding '$encoding' not supported.")
+      case "raw"  => nullCompressor
+      case "jpeg" => new JpegCompressor
+      case "compressed_segmentation" =>
+        new CompressedSegmentationCompressor(dataType,
+                                             chunkSize,
+                                             compressedSegmentationBlockSize.getOrElse(Array(8, 8, 8)))
```

Comment:
Is this block size of 8,8,8 documented somewhere as the default? Maybe add a link to the spec?

Reply:
edit: found the 8,8,8 again as privat val in CompressedSegmentation.scala – maybe make it not private and access it here? Or pass None here, and then default to the default in the implementation? The duplication may lead to problems

Reply:
There is no word in the standard about a default block size. It is the default in the janelia-flyem implementation "typically (8,8,8)". The default parameter in CompressedSegmentation/decompress could be removed, as it is not used anyway. However I would like to have some value here and not an option. This is a case that can not happen in validly formed json files anyway as block size is a required field, so I would semantically like to use `get` here, which is of course verboten.